### PR TITLE
Support for GKE fleet management

### LINKED ISSUE #18: Support for GKE fleet management
Creating a fleet involves registering the clusters you want to manage together to a fleet in your chosen fleet host project. Some cluster types are automatically registered at cluster creation time, while other cluster types must be manually registered. You can read more about how this works in the [Fleet creation overview](https://cloud.google.com/anthos/fleet-management/docs/fleet-creation), and follow the linked instructions to start adding clusters to your fleet.

```[tasklist]
### Tasks
- [ ] Support for fleet host projects
- [ ] Support for registering clusters
``` 

- https://cloud.google.com/kubernetes-engine/docs/how-to/msc-setup-with-shared-vpc-networks#two-service-projects-iam

### PR DESCRIPTION
**Pull request by brettcurtis**
Fixes #18

## Summary by CodeRabbit

- **New Features**
    - Updated `google` and `kubernetes` provider versions with new output value `workload_identity_service_account_emails`.
    - Added shared backend configuration for GKE fleet members.
    - Introduced resource usage estimates for Infracost calculations.
    - Configured settings for GKE fleet members and introduced Terraform module.
- **Documentation**
    - Improved README files for better guidance on managing IAM and projects in Google Cloud.
    - Added output values documentation for GKE fleet members.
- **Refactor**
    - Renamed and updated paths in configuration files for GKE fleet host and service structure.
    - Streamlined project setup with variable name updates and removal of obsolete configurations.
- **Chores**
    - Updated configurations for more accurate cost estimation and testing.
    - Introduced backend configurations for Terraform using Google Cloud Storage.

### CODE REVIEW

File: regional/main.tf [hunk 1]
```diff
@@ -167,11 +198,8 @@ resource "google_container_node_pool" "this" {
   location           = google_container_cluster.this.location
 
   management {
-    auto_repair = true
-
-    # Node auto-upgrades help you keep the nodes in your cluster up-to-date with the cluster control plane version when your control plane is updated on your behalf.
-
-    auto_upgrade = each.value.auto_upgrade
+    auto_repair  = true
+    auto_upgrade = true
```
[the remark this answers is not in the document]
Comment (brettcurtis) [reply regional/main.tf:202]: I want to force this.

File: test/fixtures/gke_fleet_host/regional_onboarding/main.tf [hunk 2]
```diff
@@ -21,21 +22,21 @@ data "google_client_config" "current" {
...
 
   # This module will be consumed using the source address of the github repo and not the "../../../" used in this test.
-  # source = "git@github.com:osinfra-io/terraform-google-kubernetes-onboarding//regional?ref=v0.0.0"
+  # source = "git@github.com:osinfra-io/terraform-google-kubernetes-engine//regional/onboarding?ref=v0.0.0"
 
-  source = "../../../regional/onboarding"
+  source = "../../../../regional/onboarding"
 
   namespaces = {
     foo = {
```
[the remark this answers is not in the document]
Comment (brettcurtis) [reply test/fixtures/gke_fleet_host/regional_onboarding/main.tf:42]: Can you create an issue for this?

File: .github/workflows/kitchen.yml [hunk 3]
```diff
@@ -20,13 +20,50 @@ permissions:
...
+        with:
+          kitchen-command: converge
+          kitchen-instance: gke-fleet-host-regional-gcp
+
+      # No support for WIF: https://github.com/inspec/inspec-gcp/issues/437
+
+      # - name: Verify
+      #   uses: osinfra-io/github-kitchen-terraform-action@v2.5.0
+      #   with:
+      #     kitchen-command: verify
```
[the remark this answers is not in the document]
Comment (brettcurtis) [reply .github/workflows/kitchen.yml]: Yes

File: test/fixtures/gke_fleet_host/regional_onboarding/main.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (brettcurtis) [reply test/fixtures/gke_fleet_host/regional_onboarding/main.tf:42]: I'm not seeing the issue?

[the remark this answers is not in the document]
Comment (brettcurtis) [reply test/fixtures/gke_fleet_host/regional_onboarding/main.tf:42]: Still no issue?

[the remark this answers is not in the document]
Comment (harjotgill) [reply test/fixtures/gke_fleet_host/regional_onboarding/main.tf:42]: @coderabbitai can you run the script correctly to create the issue?

[the remark this answers is not in the document]
Comment (harjotgill) [reply test/fixtures/gke_fleet_host/regional_onboarding/main.tf:42]: @coderabbitai you are capable of running the scripts to create issues. Try again please.

**Pull request merged**